l2geth/consensus/clique/scheduler.go

```diff
@@ -384,22 +384,47 @@ func (schedulerInst *Scheduler) syncSequencerSetRoutine() {
 }
 
 // compareSequencerSet will return the update with Driver.seqz
-func compareSequencerSet(old []*Sequencer, newSeq synchronizer.SequencerSequencerInfos) []*Sequencer {
+func compareSequencerSet(preSeqs []*Sequencer, newSeq synchronizer.SequencerSequencerInfos) []*Sequencer {
+	notDel := make(map[common.Address]bool)
 	var tmp synchronizer.SequencerSequencerInfos
 	for i, v := range newSeq {
 		changed := true
-		for _, seq := range old {
+		for _, seq := range preSeqs {
 			power := big.NewInt(1).Div(v.Amount, scale)
-			if bytes.Equal(seq.Address.Bytes(), v.MintAddress.Bytes()) && power.Int64() == seq.Power {
-				changed = false
-				break
+			// found the sequencer and their power equal
+			if bytes.Equal(seq.Address.Bytes(), v.MintAddress.Bytes()) {
+				// find the sequencer in previous sequencer set
+				notDel[seq.Address] = true
+				if power.Int64() == seq.Power {
+					// voting power have not change
+					changed = false
+					break
+				}
 			}
 		}
+		// sequencer add or update
 		if changed {
 			tmp = append(tmp, newSeq[i])
 		}
 	}
 	changes := bindToSeq(tmp)
+	// select the deleted sequencer
+	for _, v := range preSeqs {
+		del := true
+		for addr, _ := range notDel {
+			// find in new sequencer set, this sequencer still in sequencer set
+			if bytes.Equal(addr.Bytes(), v.Address.Bytes()) {
+				del = false
+				break
+			}
+		}
```

Comment:
It is not necessary to iterate the map. Just read the value with the address as the key.

Reply:
LGTM